### PR TITLE
App Features

### PR DESCRIPTION
**Pull request by Efiamotu-1**
### ACTIVITIES DONE FOR COMPLETE THE REQUIREMENTS
 ### Project requirements
 #### Design
 * You should follow these [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), including:
   
   * Colors.
   * Typography: font face, size and weight.
   * Layout: composition and space between elements.
 
 NOTE: In these design guidelines there are several UIs that you won't need for this exercise; also, some pages are not given a design and you will create them following the design guidelines of the other pages.
 
 Original design idea by [Gregoire Vella on Behance](https://www.behance.net/gregoirevella).
 
 The [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/) requires that you give appropriate credit to the author. Therefore, you must do it in the README of your project.
 
 #### Interactions
 * Splash screen
   
   * A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
 * Sign up and log in pages
   
   * The user should be able to register in the app with full name, email and password (all mandatory).
   * The user can log into the app using email and password.
   * If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
 * Home page (catagories page)
  
   * When the user logs in, they are presented with the categories page.
   * For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
   * When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
   * There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
 * Transactions page
   
   * For a given category, the list of transactions is presented, ordered by the most recent.
   * At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
   * There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
   * When the user clicks on the "Back" button (<), the user navigates to the home page.
 * "Add a new category" page
   
   * The user fills out a form to create a new category, indicating their name and icon (both mandatory).
   * The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
   * When the user clicks on the "Back" button (<), the user navigates to the home page.
 * "Add a new transaction" page
   
   * The user fills out a form to create a new transaction with:
     
     * name (mandatory)
     * amount (mandatory)
     * categories (mandatory at least one)
   * The user click (or taps) the "Save" button to create the new transation, and is taken to the transactions page for that category.
   * When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
 
 #### Testing requirements
 * Create unit and integration tests for all the most important components of your RoR application.
 
 #### Technical requirements
 * You should use Postgres as your database.
 * You should use devise for authentication.
 * You should validate all user input to make sure that anyone with bad intentions cannot compromise your app.
 * You can use a view template engine of your choice (.erb, .slim, .haml).
 * The project should be deployed and accessible online.
 * Your database schema should reflect the structure of an ERB diagram provide

### CODE REVIEW

File: app/views/categories/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<div class="form-style mx-5">
+  <h1>New category</h1>
+  <%= render "form", category: @category %>
+
+  <br>
+
+  
+    <%= link_to "Back to categories", categories_path, class:"btn btn-primary mb-5" %>
+  
+</div>
```
Comment (DeliceLydia) [inline app/views/categories/new.html.erb]: - [x] Please use colors font, colors, etc that were used in the [design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=)
`#3778c2`, `#5fb253`,  `#434b54`

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @Efiamotu-1, 

Good job so far! There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!


## Suggested changes
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: app/views/categories/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<div class="form-style mx-5">
+  <h1>New category</h1>
+  <%= render "form", category: @category %>
+
+  <br>
+
+  
+    <%= link_to "Back to categories", categories_path, class:"btn bg-blue mb-5" %>
+  
+</div>
```
Comment (codepantha) [inline app/views/categories/new.html.erb:10]: - [x] As the previous reviewer suggested, please use colors font, colors, etc that were used in the [design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=). Please note that the form buttons for both categories and transactions should be `#5fb523`
- [x] [OPTIONAL] Please consider placing the form buttons at the bottom of the page just like in the design
## Design Guideline
image

## Your Design
image

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @Efiamotu-1 👋, 

Well done for implementing some of the required changes 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Good commit history
✔️ No linter errors

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/categories/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<div class="form-style mx-5">
+  <h1>New category</h1>
```
Comment (t-yanick) [inline app/views/categories/new.html.erb:2]: - [ ] As indicated by previous reveiwers, change the color of the buttons to match those in the design. Check screenshot below to see that you have not changed it. Make sure you deploy the change.
Screen Shot 2023-02-17 at 7 36 27 PM

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi @Efiamotu-1 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by MeqdamAlqudah — COMMENTED** [review]
Hi @Efiamotu-1  

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a PR, that doesn't contain any changes since the last review

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by uwadonat — APPROVED** [review]
Hi @Efiamotu-1 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
### Highlights
- Amazing design ✔️
- The application works as expected ✔️
- Good commit history ✔️
approval1



# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**